src/arch/x86/object/interrupt.c

```diff
@@ -175,6 +173,13 @@ exception_t Arch_decodeIRQControlInvocation(word_t invLabel, word_t length, cte_
             return EXCEPTION_SYSCALL_ERROR;
         }
 
+#ifdef CONFIG_IOMMU
+        status = vtd_inspect_irte_msi(handle, vector);
+        if (status != EXCEPTION_NONE) {
+            return status;
+        }
+        vtd_create_irte_msi(handle, vector, pci_bus, pci_dev, pci_func);
```

Comment:
This does not look like code that is legal to do in a decode function. At least `vtd_create_irte_msi` sounds like it changes state, and if `vtd_inspect_irte_msi` is just decoding and not changing state, it should also be called `decode..` or `check..`.

All state-changing code must go into `Arch_invokeIRQControl`.

Reply:
Thanks. Is this convention documented? Where can I find the documentation?

Reply:
https://docs.sel4.systems/processes/style-guide.html in the Kernel code/Error handling section has that requirement.